packages/store/src/symbols.ts

```diff
@@ -30,6 +30,7 @@ export class NgxsConfig {
   };
 
   constructor() {
+    this.developmentMode = true;
```

Comment:
I think that this should be determined by angular's dev mode instead of default to true.
See https://angular.io/api/core/isDevMode

Reply:
@markwhitfeld as long as there is some flag to unfreeze the store, this makes sense imo.

With the flag seemingly *only* freezing the store I would further float the idea of changing the name and message to `freeze` or something similar. The current name doesn't really necessarily inform the consumer what is actually taking place imo, though it does make some sense.

Would also be sure to very clearly label this breaking change very clearly in the changelog. In the future we should probably also add another entry in the docs about this. Currently only a short description in the API I believe.

Our experience is basically that I glanced this issue and decided to turn the flag on in anticipation of version 4 without really seeing the description in my editor or going out of my to search for what it meant. Only the next day did we find the `Type error`s pop in one part of our app, where we were apparently mutating a store property due to an incomplete migration to NGXS from no state management. As the error itself does not present itself as immediately pointing to this library I didn't connect it immediately, primarily on my radar because I knew what had been changed for it to break.

So as is intended I believe this will have the intended effect of waking some up to unwarranted mutations of the state.